store/auth.go

```diff
@@ -166,3 +173,130 @@ func DischargeAuthCaveat(username, password, macaroon, otp string) (string, erro
 	}
 	return responseData.Macaroon, nil
 }
+
+func requestDeviceNonce() ([]byte, error) {
```

Comment:
The functionality in the store package is another good candidate for an isolated branch we can iterate on easily.